src/functions.php

```diff
@@ -12,7 +12,7 @@ function timeout(PromiseInterface $promise, $time, LoopInterface $loop)
     // cancelling this promise will only try to cancel the input promise,
     // thus leaving responsibility to the input promise.
     $canceller = null;
-    if ($promise instanceof CancellablePromiseInterface) {
+    if ($promise instanceof CancellablePromiseInterface  || \method_exists($promise, 'cancel') === true) {
```

Comment:
Might as well get rid of the `instanceof` or use the more explicit `$promise instanceof CancellablePromiseInterface  || (!interface_exists('React\Promise\CancellablePromiseInterface') && \method_exists($promise, 'cancel')`. What do you think about this? (see also below)

Reply:
Done

Reply:
Minor nit: `\` missing :+1:

Reply:
Will fix, lazy copy pasted it 🤐

Reply:
Fixed it